src/query/functions/src/scalars/datetime.rs

```diff
@@ -339,8 +338,19 @@ fn register_timestamp_to_date(registry: &mut FunctionRegistry) {
         val: ValueRef<TimestampType>,
         ctx: &mut EvalContext,
     ) -> Value<DateType> {
-        vectorize_with_builder_1_arg::<TimestampType, DateType>(|val, output, _| {
-            output.push(microseconds_to_days(val));
+        vectorize_with_builder_1_arg::<TimestampType, DateType>(|val, output, ctx| {
+            let tz = ctx.func_ctx.tz.tz;
+            let res = val
+                .to_timestamp(tz)
+                .naive_local()
+                .signed_duration_since(
+                    NaiveDate::from_ymd_opt(1970, 1, 1)
+                        .unwrap()
+                        .and_hms_opt(0, 0, 0)
+                        .unwrap(),
+                )
+                .num_days() as i32;
+            output.push(res);
```

Comment:
I think the logic here is right. We only consider the time zone when display.

Reply:
For a timestamp expression, the date from the timestamp. So we need  check tz when to_date(timestamp).

https://docs.snowflake.com/en/sql-reference/functions/to_date

If not , The following SQL calculation result is wrong.
```
set timezone='Asia/Shanghai';
select to_date(datetime) as day,
       sum(case when behavior_type = 'pv' then 1 else 0 end) as pv,
       count(distinct user_id) as uv
from user_behavior
group by day
order by day;
```